components/team_controller/actions/index.test.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ChannelTypes} from 'mattermost-redux/action_types';
+import * as channelsActions from 'mattermost-redux/actions/channels';
+import * as teamsActions from 'mattermost-redux/actions/teams';
+import * as teamActions from 'actions/team_actions';
+
+import mockStore from 'tests/test_store';
+import {TestHelper} from 'utils/test_helper';
+
+import {initializeTeam, joinTeam} from 'components/team_controller/actions';
+
+describe('components/team_controller/actions', () => {
+    const testUserId = 'test_user_id';
+    const testUser = TestHelper.getUserMock({id: testUserId});
+    const testTeamId = 'test_team_id';
+    const testTeam = TestHelper.getTeamMock({id: testTeamId});
+    const initialState = {
+        entities: {
+            users: {
+                profiles: {
+                    [testUserId]: testUser,
+                },
+                currentUserId: testUserId,
+            },
+            general: {
+                license: {},
+            },
+        },
+        requests: {
+            channels: {
+                getChannelsMembersCategories: {
+                    status: 'not_started',
+                    error: null,
+                },
+            },
+        },
+    };
+
+    describe('initializeTeam', () => {
```

Comment:
📝 Testing actions is a lot cleaner way to test then shallow rending the component and checking props calls. We are moving away from enzyme methodology of testing

Reply:
❤️ Great and clean tests below